transformer/classes/dockerfile/pythondockerfilegenerator.go

```diff
@@ -0,0 +1,248 @@
+	for _, mainPythonFilePath := range mainPythonFilesPath {
+		if mainPythonFileRelPath, err := filepath.Rel(baseDir, mainPythonFilePath); err == nil {
+			mainPythonFilesRelPath = append(mainPythonFilesRelPath, mainPythonFileRelPath)
+		}
+	}
+	return qaengine.FetchSelectAnswer(common.ConfigServicesKey+common.Delim+serviceName+common.Delim+common.ConfigMainPythonFileForServiceKeySegment, fmt.Sprintf("Select the main file to be used for the service %s :", serviceName), []string{fmt.Sprintf("Selected main file will be used for the service %s", serviceName)}, mainPythonFilesRelPath[0], mainPythonFilesRelPath)
+}
+
+// DirectoryDetect runs detect in each sub directory
+func (t *PythonDockerfileGenerator) DirectoryDetect(dir string) (services map[string][]transformertypes.Artifact, err error) {
+
+	pythonFiles, err := common.GetFilesByExtInCurrDir(dir, []string{pythonExt})
+	if err != nil {
+		logrus.Errorf("Error while finding python files %s", err)
+		return nil, nil
+	}
+	if len(pythonFiles) == 0 {
+		return nil, nil
+	}
+	mainPythonFilesPath, err := findMainScripts(pythonFiles)
+	if err != nil {
+		return nil, err
+	}
+	if len(mainPythonFilesPath) != 0 {
+		var serviceName, requirementsTxtPath string
+		var isDjangoProject bool
+		var detectedPorts []int32
+		detectedPorts = append(detectedPorts, 8080) //TODO: Write parser to parse and identify port
+		requirementsTxtFiles, err := common.GetFilesByName(dir, []string{requirementsTxtFile}, nil)
+		if err != nil {
+			logrus.Debugf("Cannot get the requirements.txt file: %s", err)
+		}
+		if len(requirementsTxtFiles) == 1 {
+			requirementsTxt, err := filepath.Rel(dir, requirementsTxtFiles[0])
+			if err != nil {
+				logrus.Errorf("Error in getting the relative path %s", err)
+			} else {
+				requirementsTxtPath = requirementsTxt
+				isDjangoProject = findDjangoDependency(requirementsTxtFiles[0])
+			}
+		}
+		services = map[string][]transformertypes.Artifact{
+			serviceName: {{
+				Paths: map[string][]string{
+					artifacts.ProjectPathPathType: {dir},
+					MainPythonFilesPathType:       mainPythonFilesPath,
+				}, Configs: map[string]interface{}{
+					DjangoProjectConfigType:   isDjangoProject,
+					RequirementsTxtConfigType: requirementsTxtPath,
+					DetectedPortsConfigType:   detectedPorts,
+				},
+			}},
+		}
+		return services, nil
+	}
+	return nil, nil
+}
+
+// Transform transforms the artifacts
+func (t *PythonDockerfileGenerator) Transform(newArtifacts []transformertypes.Artifact, oldArtifacts []transformertypes.Artifact) ([]transformertypes.PathMapping, []transformertypes.Artifact, error) {
+	pathMappings := []transformertypes.PathMapping{}
+	artifactsCreated := []transformertypes.Artifact{}
+	for _, a := range newArtifacts {
+		relSrcPath, err := filepath.Rel(t.Env.GetEnvironmentSource(), a.Paths[artifacts.ProjectPathPathType][0])
+		if err != nil {
+			logrus.Errorf("Unable to convert source path %s to be relative : %s", a.Paths[artifacts.ProjectPathPathType][0], err)
```

Comment:
Shouldn't we continue to the next artifact here?

Reply:
Yes sure, done. I think same thing needs to be done in other transformers in the `dockerfile` folder too?

Reply:
Actually remove the continue for now. We can consider this later.. Have to think through all scenarios, it might end up with error in other places, but that's fine. In most cases, this situation should not occur.